gdal/frmts/netcdf/netcdfsg.cpp

```diff
@@ -0,0 +1,1047 @@
+			while(nc_get_var1_int(ncId, ir_vid, &bound, &buf) == NC_NOERR)
+			{
+				bool store = buf == 0 ? false : true;
+				this->int_rings.push_back(store);
+				bound++;
+			}
+		}
+
+		delete[] nc_name;
+		delete[] pnc_name;
+		delete[] inter_name;
+		delete[] ir_name;
+
+		/* Enforcement of well formed CF files
+		 * If these are not met then the dataset is malformed and will halt further processing of
+		 * simple geometries. (Slightly permissive, letts other netCDF items to be processed)
+		 */
+
+		// part node count exists only when node count exists
+		if(pnode_counts.size() > 0 && node_counts.size() == 0)
+		{
+			throw new SG_Exception_Dep(container_name, CF_SG_PART_NODE_COUNT, CF_SG_NODE_COUNT);
+		}
+
+		// interior rings only exist when part node counts exist
+		if(int_rings.size() > 0 && pnode_counts.size() == 0)
+		{
+			throw new SG_Exception_Dep(container_name, CF_SG_INTERIOR_RING, CF_SG_PART_NODE_COUNT);
+		}	
+
+	
+		// cardinality of part_node_counts == cardinality of interior_ring (if interior ring > 0)
+		if(int_rings.size() > 0)
+		{
+			if(int_rings.size() != pnode_counts.size())
+			{
+				throw new SG_Exception_Dim_MM(container_name, CF_SG_INTERIOR_RING, CF_SG_PART_NODE_COUNT);
+			}
+		}
+
+		// lines and polygons require node_counts, multipolygons are checked with part_node_count
+		if(this->type == POLYGON || this->type == LINE)
+		{
+			if(node_counts.size() < 1)
+			{
+				throw new SG_Exception_Existential(container_name, CF_SG_NODE_COUNT);
+			}
+		}
+
+		/* Basic Safety checks End
+		 */
+
+		// Create bound list
+		int rc = 0;
+		bound_list.push_back(0);// start with 0
+
+		if(node_counts.size() > 0)
+		{
+			for(size_t i = 0; i < node_counts.size() - 1; i++)
+			{
+				rc = rc + node_counts[i];
+				bound_list.push_back(rc);	
+			}
+		}
+
+		// Node Coordinates
+		if((cart = attrf(ncId, geoVarId, CF_SG_NODE_COORDINATES)) == nullptr)
+		{
+			throw new SG_Exception_Existential(container_name, CF_SG_NODE_COORDINATES);
+		}	
+
+		// Create parts count list and an offset list for parts indexing	
+		if(this->node_counts.size() > 0)
+		{
+			// to do: check for out of bounds
+			int ind = 0; int parts = 0; int prog = 0; int c = 0;
+			for(size_t pcnt = 0; pcnt < pnode_counts.size() ; pcnt++)
+			{
+				if(prog == 0) pnc_bl.push_back(pcnt);
+
+				if(int_rings.size() > 0) if(!int_rings[pcnt]) c++;
+
+				prog = prog + pnode_counts[pcnt];
+				parts++;
+
+				if(prog == node_counts[ind])
+				{
+					ind++;
+					this->parts_count.push_back(parts);
+					if(int_rings.size() > 0) this->poly_count.push_back(c);
+					c = 0;
+					prog = 0; parts = 0;
+				}	
+				else if(prog > node_counts[ind])
+				{
+					delete[] cart;
+					throw new SG_Exception_BadSum(container_name, CF_SG_PART_NODE_COUNT, CF_SG_NODE_COUNT);
+				}
+			} 
+		}
+
+		// (1) the touple order for a single point
+		// (2) the variable ids with the relevant coordinate values
+		char * dim = strtok(cart,  " ");
+		int axis_id = 0;
+		
+		while(dim != NULL)
```

Comment:
all use of NULL to be replaced by nullptr

Reply:
sure can do that :+1: will fix